Description: Patch the Makefiles
 This patch makes the Makefiles work correctly with LDFLAGS, Debian PREFIX usages, etc.
Author: Steffen Joeris <white@debian.org>
Author: Luca Capello <luca@pca.it>
Author: Didier Raboud <odyx@debian.org>
Last-Update: 2014-01-07

--- a/Makefile
+++ b/Makefile
@@ -17,9 +17,7 @@
 VERSION=0.0
 
 # Installation prefix...
-PREFIX=/usr/local
 PREFIX=/usr
-PREFIX=$(DESTDIR)/usr
 
 # Pathnames for this package...
 BIN=$(PREFIX)/bin
@@ -38,7 +36,7 @@
 ROOT=root
 
 # Pathnames for referenced packages...
-FOODB=$(DESTDIR)/usr/share/foomatic/db/source
+FOODB=$(PREFIX)/share/foomatic/db/source
 
 # User ID's
 LPuid=-oroot
@@ -451,32 +449,32 @@
 
 
 foo2zjs: foo2zjs.o $(LIBJBG)
-	$(CC) $(CFLAGS) -o $@ foo2zjs.o $(LIBJBG)
+	$(CC) $(CFLAGS) -o $@ foo2zjs.o $(LIBJBG) $(LDFLAGS)
 
 foo2hp: foo2hp.o $(LIBJBG)
-	# $(CC) $(CFLAGS) -o $@ foo2hp.o $(LIBJBG) /usr/local/lib/libdmalloc.a
-	$(CC) $(CFLAGS) -o $@ foo2hp.o $(LIBJBG)
+	# $(CC) $(CFLAGS) -o $@ foo2hp.o $(LIBJBG) $(LDFLAGS) /usr/local/lib/libdmalloc.a
+	$(CC) $(CFLAGS) -o $@ foo2hp.o $(LIBJBG) $(LDFLAGS)
 
 foo2xqx: foo2xqx.o $(LIBJBG)
-	$(CC) $(CFLAGS) -o $@ foo2xqx.o $(LIBJBG)
+	$(CC) $(CFLAGS) -o $@ foo2xqx.o $(LIBJBG) $(LDFLAGS)
 
 foo2lava: foo2lava.o $(LIBJBG)
-	$(CC) $(CFLAGS) -o $@ foo2lava.o $(LIBJBG)
+	$(CC) $(CFLAGS) -o $@ foo2lava.o $(LIBJBG) $(LDFLAGS)
 
 foo2qpdl: foo2qpdl.o $(LIBJBG)
-	$(CC) $(CFLAGS) -o $@ foo2qpdl.o $(LIBJBG)
+	$(CC) $(CFLAGS) -o $@ foo2qpdl.o $(LIBJBG) $(LDFLAGS)
 
 foo2oak: foo2oak.o $(LIBJBG)
-	$(CC) $(CFLAGS) -o $@ foo2oak.o $(LIBJBG)
+	$(CC) $(CFLAGS) -o $@ foo2oak.o $(LIBJBG) $(LDFLAGS)
 
 foo2slx: foo2slx.o $(LIBJBG)
-	$(CC) $(CFLAGS) -o $@ foo2slx.o $(LIBJBG)
+	$(CC) $(CFLAGS) -o $@ foo2slx.o $(LIBJBG) $(LDFLAGS)
 
 foo2hiperc: foo2hiperc.o $(LIBJBG)
-	$(CC) $(CFLAGS) -o $@ foo2hiperc.o $(LIBJBG)
+	$(CC) $(CFLAGS) -o $@ foo2hiperc.o $(LIBJBG) $(LDFLAGS)
 
 foo2hbpl2: foo2hbpl2.o $(LIBJBG)
-	$(CC) $(CFLAGS) -o $@ foo2hbpl2.o $(LIBJBG)
+	$(CC) $(CFLAGS) -o $@ foo2hbpl2.o $(LIBJBG) $(LDFLAGS)
 
 
 foo2zjs-wrapper: foo2zjs-wrapper.in Makefile
@@ -556,46 +554,46 @@
 endif
 
 ok: ok.o $(LIBJBG)
-	$(CC) $(CFLAGS) ok.o $(LIBJBG) -o $@
+	$(CC) $(CFLAGS) ok.o $(LIBJBG) -o $@ $(LDFLAGS)
 
 gipddecode: gipddecode.o $(LIBJBG)
-	$(CC) $(CFLAGS) gipddecode.o $(LIBJBG) -o $@
+	$(CC) $(CFLAGS) gipddecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
 hbpldecode: hbpldecode.o $(LIBJBG)
-	$(CC) $(CFLAGS) hbpldecode.o $(LIBJBG) -o $@
+	$(CC) $(CFLAGS) hbpldecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
 hipercdecode: hipercdecode.o $(LIBJBG)
-	$(CC) $(CFLAGS) hipercdecode.o $(LIBJBG) -o $@
+	$(CC) $(CFLAGS) hipercdecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
 lavadecode: lavadecode.o $(LIBJBG)
-	$(CC) $(CFLAGS) lavadecode.o $(LIBJBG) -o $@
+	$(CC) $(CFLAGS) lavadecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
 oakdecode: oakdecode.o $(LIBJBG)
-	$(CC) $(CFLAGS) -g oakdecode.o $(LIBJBG) -o $@
+	$(CC) $(CFLAGS) -g oakdecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
 opldecode: opldecode.o $(LIBJBG)
-	$(CC) $(CFLAGS) -g opldecode.o $(LIBJBG) -o $@
+	$(CC) $(CFLAGS) -g opldecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
 qpdldecode: qpdldecode.o $(LIBJBG)
-	$(CC) $(CFLAGS) qpdldecode.o $(LIBJBG) -o $@
+	$(CC) $(CFLAGS) qpdldecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
 splcdecode: splcdecode.o $(LIBJBG)
-	$(CC) $(CFLAGS) splcdecode.o $(LIBJBG) -lz -o $@
+	$(CC) $(CFLAGS) splcdecode.o $(LIBJBG) -lz -o $@ $(LDFLAGS)
 
 slxdecode: slxdecode.o $(LIBJBG)
-	$(CC) $(CFLAGS) slxdecode.o $(LIBJBG) -o $@
+	$(CC) $(CFLAGS) slxdecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
 xqxdecode: xqxdecode.o $(LIBJBG)
-	$(CC) $(CFLAGS) xqxdecode.o $(LIBJBG) -o $@
+	$(CC) $(CFLAGS) xqxdecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
 zjsdecode: zjsdecode.o $(LIBJBG)
-	$(CC) $(CFLAGS) zjsdecode.o $(LIBJBG) -o $@
+	$(CC) $(CFLAGS) zjsdecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
 command2foo2lava-pjl: command2foo2lava-pjl.o
-	$(CC) $(CFLAGS) -L/usr/local/lib command2foo2lava-pjl.o -lcups -o $@
+	$(CC) $(CFLAGS) -L/usr/local/lib command2foo2lava-pjl.o -lcups -o $@ $(LDFLAGS)
 
 command2foo2lava-pjl.o: command2foo2lava-pjl.c
-	$(CC) $(CFLAGS) -I/usr/local/include -c command2foo2lava-pjl.c
+	$(CC) $(CPPFLAGS) $(CFLAGS) -I/usr/local/include -c command2foo2lava-pjl.c
 
 #
 # Installation rules
@@ -641,8 +639,6 @@
 	#
     
 
-UDEVBIN=$(DESTDIR)/bin/
-
 install-prog:
 	#
 	# Install driver, wrapper, and development tools
@@ -650,8 +646,7 @@
 	$(INSTALL) -d $(BIN)
 	$(INSTALL) -c $(PROGS) $(SHELLS) $(BIN)/
 	if [ "$(BINPROGS)" != "" ]; then \
-	    $(INSTALL) -d $(UDEVBIN); \
-	    $(INSTALL) -c $(BINPROGS) $(UDEVBIN); \
+	    $(INSTALL) -c $(BINPROGS) $(BIN); \
 	fi
 	#
 	# Install gamma correction files.  These are just templates,
@@ -837,10 +832,10 @@
 	    fi; \
 	done
 
-MODEL=$(DESTDIR)/usr/share/cups/model
-LOCALMODEL=$(DESTDIR)/usr/local/share/cups/model
+MODEL=$(PREFIX)/share/cups/model
+LOCALMODEL=$(PREFIX)/local/share/cups/model
 MACMODEL=/Library/Printers/PPDs/Contents/Resources
-PPD=$(DESTDIR)/usr/share/ppd
+PPD=$(PREFIX)/share/ppd
 VARPPD=/var/lp/ppd
 install-ppd:
 	#
@@ -898,9 +893,9 @@
 	    done; \
 	fi
 
-APPL=$(DESTDIR)/usr/share/applications
-OLDAPPL=$(DESTDIR)/usr/share/gnome/apps/System
-PIXMAPS=$(DESTDIR)/usr/share/pixmaps
+APPL=$(PREFIX)/share/applications
+OLDAPPL=$(PREFIX)/share/gnome/apps/System
+PIXMAPS=$(PREFIX)/share/pixmaps
 
 install-desktop:
 	#
@@ -1034,6 +1029,7 @@
 
 install-filter:
 	if [ "$(CUPS_SERVERBIN)" != "" ]; then \
+	    mkdir -p $(CUPS_SERVERBIN)/filter/; \
 	    ln -sf $(BIN)/command2foo2lava-pjl $(CUPS_SERVERBIN)/filter/; \
 	fi
 
@@ -1436,10 +1432,9 @@
 	    *)                  driver=foo2zjs;; \
 	    esac; \
 	    echo $$driver - $$printer; \
-	    ENGINE=../foomatic/foomatic-db-engine; \
 	    PERL5LIB=$$ENGINE/lib \
 		FOOMATICDB=foomatic-db \
-		$$ENGINE/foomatic-ppdfile \
+		/usr/bin/foomatic-ppdfile \
 		-d $$driver -p $$printer \
 		> PPD/$$printer.ppd; \
 	done
@@ -1873,7 +1868,7 @@
 misc: pksm2bitcmyk phorum-logo.gif
 
 pksm2bitcmyk: pksm2bitcmyk.c
-	$(CC) $(CFLAGS) pksm2bitcmyk.c -lnetpbm -o $@
+	$(CC) $(CFLAGS) pksm2bitcmyk.c -lnetpbm -o $@ $(LDFLAGS)
 
 phorum-logo.gif: archhp.fig
 	fig2dev -L gif -m.25 archhp.fig | giftrans -t "#ffffff" -o $@
--- a/icc2ps/Makefile
+++ b/icc2ps/Makefile
@@ -9,7 +9,7 @@
 LIB=	cmscam97.c cmscnvrt.c cmserr.c cmsgamma.c cmsgmt.c cmsintrp.c cmsio1.c \
 	cmslut.c cmsmatsh.c cmsmtrx.c cmsnamed.c cmspack.c cmspcs.c cmsps2.c \
 	cmssamp.c cmswtpnt.c cmsxform.c cmsio0.c cmsvirt.c
-CFLAGS=	-O3
+CFLAGS+=	-O3
 INSTALL=install
 ifeq ($(UNAME),SunOS)
     INSTALL=/usr/ucb/install
@@ -21,7 +21,7 @@
 all: foo2zjs-icc2ps man
 
 foo2zjs-icc2ps: $(SRC) $(LIB)
-	$(CC) $(CFLAGS) $(SRC) $(LIB) -lm -o $@
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(SRC) $(LIB) -lm -o $@ $(LDFLAGS)
 
 install: all
 	$(INSTALL) -c foo2zjs-icc2ps $(BIN)
